launch-scripts/launch_boot_tests_gem5_20.py

```diff
@@ -0,0 +1,157 @@
+)
+
+m5_binary = Artifact.registerArtifact(
+    command = 'make -f Makefile.x86',
+    typ = 'binary',
+    name = 'm5',
+    path =  'gem5/util/m5/m5',
+    cwd = 'gem5/util/m5',
+    inputs = [gem5_repo,],
+    documentation = 'm5 utility'
+)
+
+disk_image = Artifact.registerArtifact(
+    command = './packer build boot-exit/boot-exit.json',
+    typ = 'disk image',
+    name = 'boot-disk',
+    cwd = 'disk-image',
+    path = 'disk-image/boot-exit/boot-exit-image/boot-exit',
+    inputs = [packer, experiments_repo, m5_binary,],
+    documentation = 'Ubuntu with m5 binary installed and root auto login'
+)
+
+gem5_binary = Artifact.registerArtifact(
+    command = '''cd gem5;
+    git checkout 003c08418f841e6697b1b;
+    scons build/X86/gem5.opt -j8
+    ''',
+    typ = 'gem5 binary',
+    name = 'gem5',
+    cwd = 'gem5/',
+    path =  'gem5/build/X86/gem5.opt',
+    inputs = [gem5_repo,],
+    documentation = 'gem5 binary based on googlesource/release-staging-v20.0.0.0 (Nov 4, 2020)'
+)
+
+gem5_binary_MESI_Two_Level = Artifact.registerArtifact(
+    command = '''cd gem5;
+    git checkout d40f0bc579fb8b10da7181;
+    scons build/X86_MESI_Two_Level/gem5.opt --default=X86 PROTOCOL=MESI_Two_Level SLICC_HTML=True -j8
+    ''',
+    typ = 'gem5 binary',
+    name = 'gem5',
+    cwd = 'gem5/',
+    path =  'gem5/build/X86_MESI_Two_Level/gem5.opt',
+    inputs = [gem5_repo,],
+    documentation = 'gem5 binary based on googlesource (Nov 18, 2019)'
+)
+
+linux_repo = Artifact.registerArtifact(
+    command = '''git clone https://github.com/torvalds/linux.git;
+    mv linux linux-stable''',
+    typ = 'git repo',
+    name = 'linux-stable',
+    path =  'linux-stable/',
+    cwd = './',
+    documentation = 'linux kernel source code repo from Sep 23rd'
+)
+
+linuxes = ['5.4', '4.19.83', '4.14.134', '4.9.186', '4.4.186']
+linux_binaries = {
+    version: Artifact.registerArtifact(
+                name = f'vmlinux-{version}',
+                typ = 'kernel',
+                path = f'linux-stable/vmlinux-{version}',
+                cwd = 'linux-stable/',
+                command = f'''git checkout v{version};
+                cp ../linux-configs/config.{version} .config;
+                make -j8;
+                cp vmlinux vmlinux-{version};
+                ''',
+                inputs = [experiments_repo, linux_repo,],
+                documentation = f"Kernel binary for {version} with simple "
+                                 "config file",
+            )
+    for version in linuxes
+}
+
+if __name__ == "__main__":
+    boot_types = ['init', 'systemd']
+    num_cpus = ['1', '2', '4', '8']
+    cpu_types = ['kvm', 'atomic', 'simple', 'o3']
+    mem_types = ['classic', 'MI_example', 'MESI_Two_Level']
+
+    for linux in linuxes:
+        for boot_type in boot_types:
+            for cpu in cpu_types:
+                for num_cpu in num_cpus:
+                    for mem in mem_types:
+                        if mem == 'MESI_Two_Level':
```

Comment:
Why is MESI_Two_Level different?

Reply:
The binary location (and binary artifacts) are different.